### ISSUE TITLE
[BUG] cmake fails to build GPU package with CUDA 11

### ISSUE THREAD

**Issue by rcruzsilva** [lammps/lammps#2235]
**Summary**

Compilation of GPU package with cmake halts because nvcc compiler in cuda 11 dropped support for the sm_30 architecture. 

**LAMMPS Version and Platform**
Lammps stable version 3-Mar-2020
CUDA 11.0, Nvidia driver 450
Ubuntu 20.04, 
GeForce RTX 2080 Ti  (sm_75)

**Expected Behavior**
A successful compilation. So far did not have problem until CUDA 11. Previously compiled with CUDA 10.2 without any problem.

**Actual Behavior**
nvcc halts compilation. with the following message:
nvcc fatal   : Unsupported gpu architecture 'compute_30'

see the log attached.
[failcompilationlog.txt](https://github.com/lammps/lammps/files/4925277/failcompilationlog.txt)

**Steps to Reproduce**
-Install cuda toolkit 11.0
-set the nvcc compiler to version 11.0
-Use cmake to configure and then make to compile the GPU package.

**Further Information, Files, and Links**
I suggest to modify the line 73 in /lammps/cmake/Modules/Packages/GPU.cmake
73 if(CUDA_VERSION VERSION_GREATER "4.9")
to
73 if((CUDA_VERSION VERSION_GREATER "4.9")AND (CUDA_VERSION VERSION_LESS "11.0"))

This worked by avoiding the flag "-gencode arch=compute_30,code=[sm_30,compute_30]" in the generated "cuda_compile_fatbin*" scripts. It also prevents to generate the flags for sm_35 archictecture that was deprecated in cuda 11 and will be dropped in the next release, according to cuda 11 documentation. 

-- <<< Build configuration >>>
   Build type       RelWithDebInfo
   Install path     /home/rury/.local
   Compilers and Flags:
   C++ Compiler     /usr/bin/c++
       Type         GNU
       Version      9.3.0
   C++ Flags         -O2 -g -DNDEBUG
   Defines          LAMMPS_SMALLBIG;LAMMPS_MEMALIGN=64;LAMMPS_JPEG;LAMMPS_PNG;LAMMPS_GZIP;LAMMPS_FFMPEG
-- Static library flags:  
-- Link libraries: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so;/usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so;/usr/lib/x86_64-linux-gnu/libjpeg.so;/usr/lib/x86_64-linux-gnu/libpng.so;/usr/lib/x86_64-linux-gnu/libz.so;m;gpu
-- Using MPI with headers in /usr/lib/x86_64-linux-gnu/openmpi/include/openmpi;/usr/lib/x86_64-linux-gnu/openmpi/include and /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so;/usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so
-- GPU API: CUDA
-- GPU architecture: sm_75
-- GPU precision: MIXED

**Comment by akohlmey** [comment]
This is strictly speaking not a bug but rather a request for a feature update. CUDA 11 hasn't been available at the time the CMake scripts for GPU package were updated last, so there has been no support in LAMMPS for CUDA 11 thus far. I have updated the CMake scripts (also for our new minimum required version of cmake 3.10, which allows the use of VERSION_GREATER_EQUAL, which was not available when we were trying to be compatible with CMake 2.8).
